rpc/lib/server/http_server.go

```diff
@@ -55,13 +55,14 @@ func StartHTTPServer(
 	if config.MaxOpenConnections > 0 {
 		listener = netutil.LimitListener(listener, config.MaxOpenConnections)
 	}
-
 	go func() {
-		err := http.Serve(
-			listener,
-			RecoverAndLogHandler(maxBytesHandler{h: handler, n: maxBodyBytes}, logger),
-		)
-		logger.Info("RPC HTTP server stopped", "err", err)
+		s := &http.Server{
+			Handler:        RecoverAndLogHandler(maxBytesHandler{h: handler, n: maxBodyBytes}, logger),
+			ReadTimeout:    10 * time.Second,
```

Comment:
Is 10 seconds too long?

Reply:
I'd lower both to 3s

```suggestion
			ReadTimeout:    3 * time.Second,
```